Makefile

```diff
@@ -263,10 +263,19 @@ controller-gen: ## Download controller-gen locally if necessary.
 	@test '$(shell $(CONTROLLER_GEN) --version)' = 'Version: $(controller_gen_version)' ||\
 	$(call go-get-tool,sigs.k8s.io/controller-tools/cmd/controller-gen@$(controller_gen_version))
 
+.PHONY: kustomize
 KUSTOMIZE = $(shell pwd)/bin/kustomize
+KUSTOMIZE_VERSION = v4.5.7
 kustomize: ## Download kustomize locally if necessary.
-	@test -f $(KUSTOMIZE) ||\
-	$(call go-get-tool,sigs.k8s.io/kustomize/kustomize/v4@v4.5.7)
+	@test '$(shell $(KUSTOMIZE) version | tr -d "{" | tr -d "}" |  cut -d" " -f1)' = 'Version:kustomize/$(KUSTOMIZE_VERSION)' ||\
+	{ \
+	set -e ;\
```

Comment:
This does not work in all cases, but in particular it will not help with the pipe below.

Reply:
ack